pkg/manager/volumes/delegation/azure/disk_modifier.go

```diff
@@ -0,0 +1,308 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	klog "k8s.io/klog/v2"
+	"k8s.io/utils/ptr"
+
+	"github.com/pingcap/tidb-operator/pkg/manager/volumes/delegation"
+)
+
+var defaultWaitDuration = time.Hour * 6
+
+const (
+	paramKeyThroughput = "DiskMBpsReadWrite"
+	paramKeyIOPS       = "DiskIOPSReadWrite"
+	paramKeyType       = "skuName"
+
+	maxSize = 32767 // Azure Disk max size in GiB
+	minSize = 1
+)
+
+type AzureDiskModifier struct {
+	// for unit test, add switch for fake client
+	// because we need to get subscription id from volume ID, so we cannot initialize it in constructor
+	DiskClient DiskClient
+}
+
+type DiskClient interface {
+	Get(ctx context.Context, resourceGroupName string, diskName string, options *armcompute.DisksClientGetOptions) (armcompute.DisksClientGetResponse, error)
+	BeginUpdate(ctx context.Context, resourceGroupName string, diskName string, parameters armcompute.DiskUpdate, options *armcompute.DisksClientBeginUpdateOptions) (*azruntime.Poller[armcompute.DisksClientUpdateResponse], error)
+}
+
+type Volume struct {
+	VolumeId   string
+	Size       *int32
+	IOPS       *int64
+	Throughput *int64
+	Type       string
+
+	IsCompleted bool
+	IsFailed    bool
```

Comment:
Do these statuses exist for Azure?

Reply:
I removed these statuses. It seems that Azure didn't export these phases.